egs/chime6/s5_track2/local/diarize.sh

```diff
@@ -85,29 +86,35 @@ if [ $stage -le 4 ]; then
   echo "$0: wrote RTTM to output directory ${out_dir}"
 fi
 
+hyp_rttm=${out_dir}/rttm
+
 # For scoring the diarization system, we use the same tool that was
 # used in the DIHARD II challenge. This is available at:
 # https://github.com/nryant/dscore
+# Note that the scoring takes a single reference RTTM and a single
+# hypothesis RTTM.
 if [ $stage -le 5 ]; then
   # If a reference RTTM file is not provided, we create one using the backed up
   # segments and utt2spk files in the original data directory.
-  if [ -z $ref_rttm ]; then
-    ref_rttm=data/$name/rttm
-    echo "$0: preparing ref RTTM file from segments and utt2spk"
+  if [ -z "$ref_rttm" ]; then
     steps/segmentation/convert_utt2spk_and_segments_to_rttm.py data/$name/utt2spk.bak \
-      data/$name/segments.bak $ref_rttm
+      data/$name/segments.bak data/$name/rttm
+    ref_rttm=data/$name/rttm
   fi
-  grep 'U06' $ref_rttm > ${ref_rttm}.U06
-  ref_rttm_path=$(readlink -f ${ref_rttm}.U06)
-  out_rttm_path=$(readlink -f $out_dir/rttm)
+  echo "Diarization results for "${name}
   if ! [ -d dscore ]; then
     git clone https://github.com/nryant/dscore.git || exit 1;
     cd dscore
     python -m pip install --user -r requirements.txt
     cd ..
   fi
-  cd dscore
-  python score.py -r $ref_rttm_path -s $out_rttm_path
-  cd ..
+  sed 's/_U0[1-6]//g' $ref_rttm > $ref_rttm.scoring
+  sed 's/_U0[1-6]//g' $hyp_rttm > $hyp_rttm.scoring
```

Comment:
I suppose it should be  `'s/\.ENH//g'`

Reply:
No, this is to normalize the ref and hyp RTTM array ids. For example, the reference RTTM may have different array ids like S02_U02.ENH, S02_U03.ENH and so on, depending on which array was referenced for transcribing. But in our hyp RTTM, we may have just used U06, in which case the scoring would be thrown off. We remove array information since we want to score sessions, not arrays. This is my understanding, please correct me if I'm wrong.

Reply:
So in the `rttm.scoring` file, the file ID (e.g. S09_U01) will become S09, I see. I think we need to do the same thing for the uem file, otherwise, it can't find the scoring region, right?

Reply:
Yes, you are right. Could you make this change for the UEM file?

Reply:
Sure. But I met with such errors when I run the last step of diarize.sh. 
```
WARNING: Skipping turn from file not in UEM. TURN: FILE: S21.ENH, SPEAKER: 3, ONSET: 9199.000000, OFFSET: 9200.430000, DUR: 1.430000
WARNING: File "S01" missing in reference RTTMs.
WARNING: File "S01" missing in system RTTMs.
WARNING: File "S02" missing in reference RTTMs.
WARNING: File "S02" missing in system RTTMs.
WARNING: File "S09" missing in reference RTTMs.
WARNING: File "S09" missing in system RTTMs.
WARNING: File "S21" missing in reference RTTMs.
WARNING: File "S21" missing in system RTTMs.
WARNING: No reference speaker turns found within UEM scoring regions.
WARNING: No system speaker turns found within UEM scoring regions.
Traceback (most recent call last):
  File "score.py", line 311, in <module>
    main()
  File "score.py", line 305, in main
    ignore_overlaps=args.ignore_overlaps)
  File "/scratch/near/espnet/tools/kaldi/egs/chime6/s5_track2/dscore/scorelib/score.py", line 232, in score
    ref_turns, sys_turns, uem=uem, **kwargs)
  File "/scratch/near/espnet/tools/kaldi/egs/chime6/s5_track2/dscore/scorelib/metrics.py", line 503, in der
    global_der = file_to_der_base['ALL']
KeyError: u'ALL'
```
If I remove the uem option from score.py, it works well. Could you help debug it? Thanks.

Reply:
Yeah the problem is that the UEM file does not have the ".ENH" suffix, so we would need to remove this from the the ref and hyp rttm as well for scoring. You can just replace the `s/_U0[1-6]//g` with `s/_U0[1-6].ENH//g` and it should work.

Reply:
Also, after removing the array ids from the UEM file, you will have multiple entries corresponding to each session, which will again throw an error in scoring. So you'll need to do something like:
```
cat ./local/uem_file | grep 'U06' | sed 's/_U0[1-6]//g' > ./local/uem_file.tmp
```